library/src/extra/rocsparse_spgeam.cpp

```diff
@@ -92,17 +144,201 @@ namespace rocsparse
+        ROCSPARSE_CHECKARG_POINTER(2, mat_A);
+        ROCSPARSE_CHECKARG_POINTER(3, mat_B);
+        if(stage == rocsparse_spgeam_stage_compute)
+        {
+            ROCSPARSE_CHECKARG_POINTER(4, mat_C);
+        }
+        ROCSPARSE_CHECKARG_ENUM(5, stage);
+
+        ROCSPARSE_CHECKARG(2, mat_A, (mat_A->init == false), rocsparse_status_not_initialized);
+        ROCSPARSE_CHECKARG(3, mat_B, (mat_B->init == false), rocsparse_status_not_initialized);
+
+        ROCSPARSE_CHECKARG(
+            3, mat_B, (mat_B->format != mat_A->format), rocsparse_status_not_implemented);
+
+        ROCSPARSE_CHECKARG(2,
+                           mat_A,
+                           (mat_A->data_type != descr->get_compute_datatype()),
+                           rocsparse_status_not_implemented);
+        ROCSPARSE_CHECKARG(3,
+                           mat_B,
+                           (mat_B->data_type != descr->get_compute_datatype()),
+                           rocsparse_status_not_implemented);
+
+        ROCSPARSE_CHECKARG(
+            3, mat_B, (mat_B->row_type != mat_A->row_type), rocsparse_status_type_mismatch);
+        ROCSPARSE_CHECKARG(
+            3, mat_B, (mat_B->col_type != mat_A->col_type), rocsparse_status_type_mismatch);
+
+        if(stage == rocsparse_spgeam_stage_compute || mat_C != nullptr)
+        {
+            ROCSPARSE_CHECKARG(4, mat_C, (mat_C->init == false), rocsparse_status_not_initialized);
+
+            ROCSPARSE_CHECKARG(
+                4, mat_C, (mat_C->format != mat_A->format), rocsparse_status_not_implemented);
+
+            ROCSPARSE_CHECKARG(4,
+                               mat_C,
+                               (mat_C->data_type != descr->get_compute_datatype()),
+                               rocsparse_status_not_implemented);
+
+            ROCSPARSE_CHECKARG(
+                4, mat_C, (mat_C->row_type != mat_A->row_type), rocsparse_status_type_mismatch);
+            ROCSPARSE_CHECKARG(
+                4, mat_C, (mat_C->col_type != mat_A->col_type), rocsparse_status_type_mismatch);
+        }
+
+        return rocsparse_status_continue;
+    }
+
+    static rocsparse_status spgeam_checkarg(rocsparse_handle            handle, //0
+                                            rocsparse_spgeam_descr      descr, //1
+                                            const void*                 alpha, //2
+                                            rocsparse_const_spmat_descr mat_A, //3
+                                            const void*                 beta, //4
+                                            rocsparse_const_spmat_descr mat_B, //5
+                                            rocsparse_spmat_descr       mat_C, //6
+                                            rocsparse_spgeam_stage      stage, //7
+                                            size_t                      buffer_size, //8
+                                            void*                       temp_buffer) //9
```

Comment:
I think the validity of temp_buffer is missing. If buffer_size > 0, then temp_buffer must be a non-null pointer.